lib/rpmsg/rpmsg.c

```diff
@@ -271,7 +269,7 @@ struct rpmsg_endpoint *rpmsg_create_ept(struct rpmsg_device *rdev,
 			goto ns_err;
 	}
 
-	return ept;
+	return 0;
 
 ns_err:
 	rpmsg_unregister_endpoint(ept);
```

Comment:
In  case of error, shall we also need to unset the endpoint address?
Or maybe we can assume registering endpoint always succeed. as it just add to  link list